src/x509.cc

```diff
@@ -45,6 +45,67 @@ std::string parse_args(const Nan::FunctionCallbackInfo<v8::Value>& info) {
   return *String::Utf8Value(info[0]->ToString());
 }
 
+
+
+NAN_METHOD(verify) {
+  Nan::HandleScope scope;
+  OpenSSL_add_all_algorithms();
+
+  std::string cert_path = *String::Utf8Value(info[0]->ToString());
+  std::string ca_bundlestr = *String::Utf8Value(info[1]->ToString());
+  /*
+    Create x509 cert
+   */
+  X509_STORE *store = NULL;
+  X509_STORE_CTX *vrfy_ctx = NULL;
+  X509 *cert = NULL;
+  BIO *certbio = BIO_new(BIO_s_file());
+
+  if (!(store=X509_STORE_new())){
+    X509_STORE_free(store);
+    BIO_free_all(certbio);
+    Nan::ThrowError("Failed to create X509 certificate store.");
+  }
+
+  vrfy_ctx = X509_STORE_CTX_new();
+
+  // load file in BIO
+  int ret = BIO_read_filename(certbio, cert_path.c_str());
+
+  if (! (cert = PEM_read_bio_X509(certbio, NULL, 0, NULL))) {
```

Comment:
By the way, could you please separate this complex statement to clear subsequent ones like this:

```cpp
cert = PEM_read_bio_X509(cert_bio, NULL, 0, NULL);
if (!cert) {
  // ...
}
```